### PR TITLE
Hide cache methods in Configuration class

### PR DESCRIPTION
**Pull request by bzwei**
Previously the whole caches are exposed. Now only provide query methods
to access cached data (for api-version and token)

### CODE REVIEW

File: lib/azure/armrest/configuration.rb [hunk 1]
```diff
@@ -193,7 +189,10 @@ def validate_token_time(time)
       # a non-preview version that is not set in the future. Otherwise, just
       # just the most recent one.
       #
-      def set_provider_info(providers)
+      def set_provider_api_versions
+        # A lookup table for getting api-version strings per provider and service.
+        @provider_api_versions = Hash.new { |hash, key| hash[key] = {} }
```
Comment (djberg96) [inline lib/azure/armrest/configuration.rb]: Shouldn't the @provider_api_versions initialization happen in the constructor?

File: lib/azure/armrest/configuration.rb [hunk 2]
```diff
@@ -86,39 +83,51 @@ def clear_caches
...
-        # A one-time lookup table set at the class level
         @providers = fetch_providers
-        set_provider_info(@providers) if self.class.provider_version_cache.empty?
+        set_provider_api_versions
+      end
+
+      def hash
+        [tenant_id, client_id, client_key].join('_').hash
+      end
+
```
Comment (djberg96) [inline lib/azure/armrest/configuration.rb]: Should the hash method be private?

File: lib/azure/armrest/configuration.rb (hunk 1 above)
Comment (bzwei) [inline lib/azure/armrest/configuration.rb]: No need. `set_provider_info` is called by the constructor anyway. Technically if this method is called separately (which does not happen), `set_provider_info` needs to be reset here.

File: lib/azure/armrest/configuration.rb (hunk 2 above)
Comment (bzwei) [inline lib/azure/armrest/configuration.rb]: `hash` and `eql?` are twin overridden methods from `Object`, similar to `hashCode` and `equals` in Java.

File: lib/azure/armrest/configuration.rb (hunk 1 above)
Comment (djberg96) [inline lib/azure/armrest/configuration.rb]: I'm just wondering if that will cause a warning in verbose mode. We can always revisit.

**Comment by djberg96** [comment]
Looks good. I guess it would be best to fix those rubocop TZ warnings first.

**Pull request merged**